File: app/src/main/java/usage/ywb/wrapper/modules/home/entity/Domain.java
package usage.ywb.wrapper.modules.home.entity;


import android.os.Parcel;
import android.os.Parcelable;

import com.google.gson.annotations.SerializedName;

/**
 * @author yuwenbo
 * @version [ V.2.4.4  2019/3/29 ]
 */
public class Domain implements Parcelable {

    @SerializedName("DOMAIN")
    public String domain;//域名

    protected Domain(Parcel in) {
        domain = in.readString();
    }

    public static final Creator<Domain> CREATOR = new Creator<Domain>() {
        @Override
        public Domain createFromParcel(Parcel in) {
            return new Domain(in);
        }

        @Override
        public Domain[] newArray(int size) {
            return new Domain[size];
        }
    };

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeString(domain);
    }
}

File: app/src/main/java/usage/ywb/wrapper/modules/home/entity/Domain.kt
package usage.ywb.wrapper.modules.home.entity

import android.os.Parcel
import android.os.Parcelable
import com.google.gson.annotations.SerializedName

/**
 * @author yuwenbo
 * @version [ V.2.4.4  2019/3/29 ]
 */
class Domain protected constructor(`in`: Parcel) : Parcelable {
    @JvmField
    @SerializedName("DOMAIN")
    var domain //域名
            : String?

    override fun describeContents(): Int {
        return 0
    }

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeString(domain)
    }

    companion object {
        val CREATOR: Parcelable.Creator<Domain> = object : Parcelable.Creator<Domain> {
            override fun createFromParcel(`in`: Parcel): Domain? {
                return Domain(`in`)
            }

            override fun newArray(size: Int): Array<Domain?> {
                return arrayOfNulls(size)
            }
        }
    }

    init {
        domain = `in`.readString()
    }
}